### PR TITLE
Add endpoint to get UTXO set for an address

### PR DESCRIPTION
**Pull request by rowandh**
Uses `UtxoIndexer` which is slow in its current implementation because it has to enumerate the whole chain. Improvements could by made by optimizing the indexer.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs [hunk 1]
```diff
@@ -289,6 +296,64 @@ public IActionResult GetUtxoSet(int atBlockHeight)
...
+
+                    var utxoModel = new UtxoModel()
+                    {
+                        TxId = utxo.Hash,
+                        Index = utxo.N,
+                        ScriptPubKey = output.ScriptPubKey,
+                        Value = output.Value
+                    };
+
+                    utxos.Add(utxoModel);
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs:339]: I think we have to be careful about thread safety here - if the order does indeed not matter then using one of the concurrent collections instead would be preferable

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs [hunk 2]
```diff
@@ -289,6 +296,64 @@ public IActionResult GetUtxoSet(int atBlockHeight)
...
+            // Get coinview at current height (SLOW)
+            var coinView = this.utxoIndexer.GetCoinviewAtHeight(this.chainIndexer.Height);
+
+            var utxos = new List<UtxoModel>();
+
+            // Get utxos for this address
+            try
+            {
+                // Fine to do this in parallel because we don't care about the order
+                Parallel.ForEach(coinView.UnspentOutputs, (utxo) =>
```
Comment (mrtpain) [inline src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs:319]: Would it be possible/more performance to break this loop when the amount needed for a transaction is found? 

Example: I need uxtos to cover 50 CRS, only return UXTOs needed for up to that amount (maybe plus 1 for gas).

Maybe not an option or worth it, just a thought. Most users of this endpoint will only need _enough_ to cover the transaction not necessary _all_ of them.

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs (hunk 1 above)
Comment (rowandh) [reply src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs:340]: Good call, changed to use a concurrent bag.

File: src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs (hunk 2 above)
Comment (rowandh) [reply src/Stratis.Bitcoin.Features.BlockStore/Controllers/BlockStoreController.cs:320]: Definitely. That sounds like a UTXO selection endpoint which isn't the goal with this PR, but could be added after the utxo indexer has been optimized.

**Review by zeptin — APPROVED** [review]

**Pull request merged**